crates/context/src/handlers/create_context.rs

```diff
@@ -240,17 +240,14 @@ async fn create_context(
 
     let storage = ContextStorage::from(datastore, context.id);
 
+    debug!("Compiling module for context initialization");
+
     let module = engine.compile(&blob)?;
 
-    let (outcome, storage) = execute(
-        &guard,
-        identity,
-        module,
-        "init".into(),
-        init_params.into(),
-        storage,
-    )
-    .await?;
+    let mut storage_mut = storage;
+    let outcome = module.run(context.id, identity, "init", &init_params, &mut storage_mut)?;
+
+    let (outcome, storage) = (outcome, storage_mut);
```

Comment:
why this pattern? `storage` is defined 3 statements earlier, and can be moved closer if you wanted to localize the `mut` declaration, a new variable is not needed, and I don't understand `outcome = outcome`